solr/contrib/blob-directory/src/java/org/apache/solr/blob/BlobDirectory.java

```diff
@@ -61,20 +62,33 @@
   private final Object lock = new Object();
   private volatile boolean isOpen;
 
-  public BlobDirectory(Directory delegate, String blobDirPath, BlobStoreConnection blobStoreConnection) throws IOException {
+  public BlobDirectory(Directory delegate, String blobDirPath, BlobRepository blobRepository) throws IOException {
     super(delegate);
     this.blobDirPath = blobDirPath;
-    this.blobStoreConnection = blobStoreConnection;
+    this.blobRepository = blobRepository;
     pullMissingFilesFromRepo();
   }
 
   private void pullMissingFilesFromRepo() throws IOException {
     Set<String> localFileNames = new HashSet<>(Arrays.asList(in.listAll()));
```

Comment:
blobRepository.pull might not be internally multi-threaded (?) but I could easily imagine we might consider making it so.  I think we should either make localFileNames & numPulledFiles thread-safe (or use explicit synchronization; probably simpler?) OR add a comment to recognize we don't need to worry about thread-safety at this time.  This way if it happens, we'll be less likely to make a mistake then.

Reply:
I'll add both a comment and javadoc for the fileFilter param of the pull method.
Currently the file selection is done on the current thread. Then the files are pulled by multiple threads.